lib/HLSL/HLOperationLower.cpp

```diff
@@ -5671,7 +5671,7 @@ void TranslateStructBufMatSubscript(CallInst *CI, Value *handle,
     resultSize = resultType->getVectorNumElements();
   DXASSERT(resultSize <= 16, "up to 4x4 elements in vector or matrix");
   _Analysis_assume_(resultSize <= 16);
-  Value *idxList[16];
+  std::vector<Value *> idxList(resultSize);
```

Comment:
What was the issue here? Were we accessing an uninitialized value? Could still use an array and just initialize the elements to nullptr `Value *idxList[16] = {0};`.

Reply:
The problem is we need idxList has correct size which is used in HLMatrixLower::LowerGEPOnMatIndexListToIndex.